app/models/topic.rb

```diff
@@ -335,6 +335,9 @@ def recover!(recovered_by = nil)
 
   scope :exclude_scheduled_bump_topics, -> { where.not(id: TopicTimer.scheduled_bump_topics) }
 
+  scope :exclude_auto_bumped_topics,
+        -> { where.not(id: Post.where(action_code: "autobumped").select(:topic_id)) }
```

Comment:
Would like to solicit some feedback here @tgxworld.

Since we don't index `action_code`, this will cause a sequential scan on `posts` as a sub-plan, which could be pretty bad in sites with lots of posts.

**Before:**

<img width="657" alt="Screenshot 2023-01-27 at 12 53 40 PM" src="https://user-images.githubusercontent.com/5259935/215012451-a0cbb43c-7e3f-48c6-81b7-161348cd43ea.png">

**After:**

<img width="661" alt="Screenshot 2023-01-27 at 12 54 51 PM" src="https://user-images.githubusercontent.com/5259935/215012471-ba6ecb95-5b21-4b9d-819b-7e174790eafc.png">

---

There are a couple takes here:

1. This is fine, because the query is run once every 15 minutes in a background job, and we were already forcing a sequential scan on topics (although that's one to two orders of magnitude smaller than posts.) Another argument against might be someone uses this scope in a hotter context.

2. We might want to add an index on `posts.action_type`. Since most of these will be `NULL`, a partial index with condition `IS NOT NULL` might be beneficial.

3. We might want to add a cache column on `topics`. This creates a bit of extra complexity, because the cache needs to be kept up-to-date, and this is complicated by the fact that it is possible to delete auto-bump small action posts which in the current implementation makes it count as not auto-bumped.

WDYT?

Reply:
I think (2) sounds OK but instead of indexing on `posts.action_type`, I think we can index on `posts.topic_id` with a `posts.action_type = "autobumped"` condition? That would keep the index much smaller.

Reply:
Thanks, @tgxworld! This PR is on hold because we're looking to change the feature entirely, but good insight here from this discussion. 🙂